cmd/telemetrygen/internal/metrics/worker.go

```diff
@@ -28,29 +29,63 @@ type worker struct {
 
 func (w worker) simulateMetrics(res *resource.Resource, exporter sdkmetric.Exporter) {
 	limiter := rate.NewLimiter(w.limitPerSecond, 1)
-	var i int64
 
+	var i int64
 	for w.running.Load() {
-		rm := metricdata.ResourceMetrics{
-			Resource: res,
-			ScopeMetrics: []metricdata.ScopeMetrics{
-				{
-					Metrics: []metricdata.Metrics{
+		var metrics []metricdata.Metrics
+		if w.metricType == metricTypeGauge || w.metricType == metricTypeAll {
+			metrics = append(metrics, metricdata.Metrics{
+				Name: "gen.metric.gauge",
```

Comment:
Let's keep the name as `"gen"` for all metrics (this implies removing the `"all"` option). I think we should keep the name as `"gen"` because:
1. It would be a breaking change for people already using this
2. We may want to add an option to change the name in the future and it makes sense to start with the same default always

Reply:
Reverted metric name back to "gen"